salt/modules/match.py

```diff
@@ -36,18 +33,16 @@ def compound(tgt, minion_id=None):
         salt '*' match.compound 'L@cheese,foo and *'
     """
     if minion_id is not None:
-        opts = copy.copy(__opts__)
-        if not isinstance(minion_id, six.string_types):
-            minion_id = six.text_type(minion_id)
-        opts["id"] = minion_id
-    else:
-        opts = __opts__
-    matchers = salt.loader.matchers(opts)
+        if not isinstance(minion_id, str):
+            minion_id = str(minion_id)
+    matchers = salt.loader.matchers(__opts__)
     try:
-        return matchers["compound_match.match"](tgt)
+        ret = matchers["compound_match.match"](tgt, opts=__opts__, minion_id=minion_id)
```

Comment:
why do we need to pass `__opts__` here? shouldn't  `__opts__` be available in `compound_match.match`?

Reply:
Passing __opts__ is needed for metaproxy proxies.